packages/apps/reputation-oracle/server/src/modules/kyc/kyc.service.spec.ts

```diff
@@ -1,124 +1,120 @@
 import { KycEntity } from './kyc.entity';
+import { KycError, KycErrorMessage } from './kyc.error';
 import { KycRepository } from './kyc.repository';
 import { KycService } from './kyc.service';
-import { KycError, KycErrorMessage } from './kyc.error';
-import { Web3ConfigService } from '../../config/web3-config.service';
+
+const mockHttpService = {
+  post: jest.fn(),
+};
+
+const mockKycConfigService = {
+  apiKey: faker.string.alphanumeric(),
+  baseUrl: faker.internet.url(),
+};
+
+const operatorWallet = generateEthWallet();
 
 describe('Kyc Service', () => {
   let kycService: KycService;
   let httpService: HttpService;
   let kycRepository: KycRepository;
   let kycConfigService: KycConfigService;
-  let web3Service: Web3Service;
 
   beforeAll(async () => {
-    const mockHttpService: DeepPartial<HttpService> = {
-      axiosRef: {
-        request: jest.fn(),
-      },
-    };
-
-    const signerMock = {
-      address: MOCK_ADDRESS,
-      getNetwork: jest.fn().mockResolvedValue({ chainId: 1 }),
-      signMessage: jest.fn(),
-    };
-
     const moduleRef = await Test.createTestingModule({
       providers: [
-        {
-          provide: ConfigService,
-          useValue: {
-            get: jest.fn((key: string) => mockConfig[key]),
-            getOrThrow: jest.fn((key: string) => {
-              if (!mockConfig[key]) {
-                throw new Error(`Configuration key "${key}" does not exist`);
-              }
-              return mockConfig[key];
-            }),
-          },
-        },
         KycService,
-        KycConfigService,
-        HCaptchaConfigService,
-        {
-          provide: HttpService,
-          useValue: mockHttpService,
-        },
+        { provide: HttpService, useValue: mockHttpService },
+        { provide: KycConfigService, useValue: mockKycConfigService },
         { provide: KycRepository, useValue: createMock<KycRepository>() },
-        Web3ConfigService,
         {
-          provide: Web3Service,
+          provide: Web3ConfigService,
           useValue: {
-            getSigner: jest.fn().mockReturnValue(signerMock),
+            operatorAddress: operatorWallet.address,
+            privateKey: operatorWallet.privateKey,
+            network: Web3Network.TESTNET,
+            reputationNetworkChainId: ChainId.POLYGON_AMOY,
+            getRpcUrlByChainId: jest.fn().mockReturnValue(faker.internet.url()),
           },
         },
+        Web3Service,
       ],
     }).compile();
 
     httpService = moduleRef.get<HttpService>(HttpService);
     kycService = moduleRef.get<KycService>(KycService);
     kycRepository = moduleRef.get<KycRepository>(KycRepository);
     kycConfigService = moduleRef.get<KycConfigService>(KycConfigService);
-    web3Service = moduleRef.get<Web3Service>(Web3Service);
+  });
 
-    jest
-      .spyOn(KycConfigService.prototype, 'apiKey', 'get')
-      .mockReturnValue('test');
+  afterEach(() => {
+    jest.clearAllMocks();
```

Comment:
```suggestion
    jest.resetAllMocks();
```

To ensure independent test runs